src/Resources/contao/models/PageModel.php

```diff
@@ -927,12 +930,6 @@ public function loadDetails()
 
 		$this->trail = array_reverse($trail);
 
-		// Do not cache protected pages
-		if ($this->protected)
```

Comment:
Because it's wrong and not needed. Caching should not be disabled just because the page is protected. In this PR, whenever a user is logged in, the response is set to private, so it never caches.